pkg/certs/cert-inspection/certgraphanalysis/deduplication.go

```diff
@@ -56,10 +81,25 @@ func combineCertOnDiskLocations(in *certgraphapi.CertKeyPair, rhs []certgraphapi
 			}
 		}
 		if !found {
+			// Store key to be merged into Cert
+			if curr.Cert.Path == "" {
+				keyLocation = curr.Key
+				continue
+			}
 			out.Spec.OnDiskLocations = append(out.Spec.OnDiskLocations, curr)
 		}
 	}
 
+	// Fill in Key property if it was found earlier and unset
+	if len(keyLocation.Path) == 0 {
+		return out
+	}
+	for idx, loc := range out.Spec.OnDiskLocations {
+		if loc.Key.Path == "" {
+			out.Spec.OnDiskLocations[idx].Key = keyLocation
```

Comment:
I'm not clear on why we're doing this.  This appears to be saying, "every cert location that lacks a corresponding key location gets a key location from some other location pair".  But I'm not sure why we want to do that.

Reply:
we want to merge two locations if one has the cert and the other has the key (with matching `PubkeyModulus`). It also stores unmatching certs/key (needs improved test to handle PEM certkey pair)